cloudplatform/connectivity-oauth/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/IdentityAuthenticationServiceBindingDestinationLoader.java

```diff
@@ -0,0 +1,318 @@
+import com.sap.cloud.sdk.cloudplatform.connectivity.exception.DestinationNotFoundException;
+
+import io.vavr.control.Try;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * A {@link ServiceBindingDestinationLoader} that loads destinations from service bindings backed by the Identity
+ * Authentication Service (IAS).
+ *
+ * @since 5.6.0
+ */
+@Beta
+@Slf4j
+public class IdentityAuthenticationServiceBindingDestinationLoader implements ServiceBindingDestinationLoader
+{
+    private static final Exception NOT_AN_IAS_SERVICE_BINDING =
+        new DestinationNotFoundException("The bound service is not backed by the IAS service.");
+    private static final DestinationAccessException NO_ENDPOINTS_DEFINED =
+        new DestinationAccessException("The IAS-based service binding does not contain any HTTP endpoints.");
+
+    private static final ServiceIdentifier NULL_IDENTIFIER = ServiceIdentifier.of("unknown-service");
+    @Nonnull
+    private final ServiceBindingDestinationLoader delegateLoader;
+
+    /**
+     * The default constructor.
+     */
+    public IdentityAuthenticationServiceBindingDestinationLoader()
+    {
+        this(ServiceBindingDestinationLoader.defaultLoaderChain());
+    }
+
+    // for testing purposes
+    IdentityAuthenticationServiceBindingDestinationLoader(
+        @Nonnull final ServiceBindingDestinationLoader delegateLoader )
+    {
+        this.delegateLoader = delegateLoader;
+    }
+
+    @Nonnull
+    @Override
+    public Try<HttpDestination> tryGetDestination( @Nonnull final ServiceBindingDestinationOptions options )
+    {
+        final ServiceBinding serviceBinding = options.getServiceBinding();
+        final Try<IasServiceBindingView> maybeBindingView =
+            Try.of(() -> IasServiceBindingView.fromServiceBindingOrThrow(serviceBinding));
+        if( maybeBindingView.isFailure() ) {
+            return Try.failure(maybeBindingView.getCause());
+        }
+
+        final IasServiceBindingView bindingView = maybeBindingView.get();
+        if( bindingView == null ) {
+            return Try.failure(NOT_AN_IAS_SERVICE_BINDING);
+        }
+
+        final Try<HttpEndpointEntry> maybeEndpoint = Try.of(() -> getEndpointOrThrow(bindingView));
+        if( maybeEndpoint.isFailure() ) {
+            return Try.failure(maybeEndpoint.getCause());
+        }
+
+        final HttpEndpointEntry endpoint = maybeEndpoint.get();
+
+        final ServiceBindingDestinationOptions.Builder optionsBuilder =
+            ServiceBindingDestinationOptions
+                .forService(ServiceIdentifier.of("identity"))
+                .onBehalfOf(options.getOnBehalfOf())
+                .withOption(BtpServiceOptions.IasOptions.withTargetUri(endpoint.uri));
+
+        if( !endpoint.requiresTokenForTechnicalAccess
+            && options.getOnBehalfOf() != OnBehalfOf.NAMED_USER_CURRENT_TENANT ) {
+            optionsBuilder.withOption(BtpServiceOptions.IasOptions.withMutualTlsOnly());
```

Comment:
I'm wondering whether this is correct for the "current Tenant is subscriber" case... Probably not, right?

So it should be 

```suggestion
        if( !endpoint.requiresTokenForTechnicalAccess
            && options.getOnBehalfOf() == OnBehalfOf.TECHNICAL_USER_PROVIDER ) {
            optionsBuilder.withOption(BtpServiceOptions.IasOptions.withMutualTlsOnly());
```

Reply:
As discussed, this implementation has been shifted towards the `OAuth2PropertySupplier` as this class (the `IdentityAuthenticationServiceBindingDestinationLoader`) cannot determine whether the current tenant is the provider.

Also see the changes for within the `IasOptions` class for reference.